packages/app/src/components/Root/Root.tsx

```diff
@@ -70,22 +69,30 @@ const SidebarLogo: FC<{}> = () => {
   );
 };
 
-const Root: FC<{}> = ({ children }) => (
+type SidebarItem = {
+  icon: typeof SvgIcon;
```

Comment:
`@backstage/core` exposes `IconComponent` which we would prefer to use here.

Reply:
> What are your thoughts on moving the Sidebar to a separate file inside app, so that it can be focused only on enumerating items, and then possibly moving SidebarPage to core?

That would be much cleaner, I'm guessing. Also aware of this design being subpar, but I'm just trying to optimize for not getting flooded in merge-conflicts in files that I (or whoever is configuring our deployment of the Backstage app) would have to weed through for relevant bits. Could be that my concerns are misplaced though. :shrug:

I can check off the points I raised later but perhaps we can decide now if this is maybe just a waste of time. Perhaps, dreading merges and working through the conflicts won't be as bad as I'm thinking. :wink:

Reply:
I think it's great that you worry about it - it's an important factor of how easy the solution will be to operate over time.

I think it could be helpful to clarify that in the end, we would like the app package that you maintain to be very slim. It would be mainly config, style overrides, wiring, and a thin app shell in JSX. The normal way of starting to use Backstage in an org would be to essentially run `backstage-cli create-app` or similar. Then the created package won't be pulling changes from an upstream anymore.

That would mean that the app package you see in the backstage org, would only be a "demo app" with some examples of how to perform some of those common tasks.

So even if YOUR app package contained JSX describing the contents of the sidebar, its contact surface with the open source project will be its `import`s from core. Of course that "api" may change - new props on the shared components used etc - but maybe that better explains what the upgrade experience might look like.

Reply:
Loud and clear. Let's go ahead and close this PR and I'll remember to keep an eye on Root.tsx to do whatever is needed post-merge for now.